main/src/ca/uwaterloo/flix/language/phase/typer/BaseEffects.scala

```diff
@@ -85,4 +92,100 @@ object BaseEffects {
       Type.mkUnion(tpes, loc)
   }
 
+  /**
+    * Parses a JSON file of the form:
+    *
+    * {{{
+    * {
+    *   "classes": {
+    *     "java.lang.ProcessBuilder": "Exec, FileRead",
+    *     "java.lang.reflect.Method": "Sys"
+    *   }
+    * }
+    * }}}
+    */
+  private def loadClassEffs(): Map[Class[?], Set[Symbol.EffectSym]] = {
+    val data = LocalResource.get(ClassEffsPath)
+    val json = parse(data)
+
+    val m = json \\ "classes" match {
+      case JObject(l) => l.map {
+        case (className, JString(s)) =>
+          val clazz = Class.forName(className)
+          val effSet = s.split(",").map(_.trim).map(Symbol.parseBaseEff).toSet
+          (clazz, effSet)
+        case _ => throw InternalCompilerException("Unexpected field value.", SourceLocation.Unknown)
+      }
+      case _ => throw InternalCompilerException("Unexpected JSON format.", SourceLocation.Unknown)
+    }
+
+    m.toMap
+  }
+
+  /**
+    * Parses a JSON file of the form:
+    *
+    * {{{
+    * {
+    *   "constructors": {
+    *     "java.lang.ProcessBuilder": "Exec",
+    *     "java.net.URL": "Net"
+    *   }
+    * }
+    * }}}
+    *
+    * Note: The effect set applies to *ALL* constructors of the class.
+    */
+  private def loadConstructorEffs(): Map[Constructor[?], Set[Symbol.EffectSym]] = {
+    val data = LocalResource.get(ConstructorEffsPath)
+    val json = parse(data)
+
+    val m = json \\ "constructors" match {
+      case JObject(l) => l.flatMap {
+        case (className, JString(s)) =>
+          val clazz = Class.forName(className)
+          val effSet = s.split(",").map(_.trim).map(Symbol.parseBaseEff).toSet
+          clazz.getConstructors.map(c => (c, effSet))
+        case _ => throw InternalCompilerException("Unexpected field value.", SourceLocation.Unknown)
+      }
+      case _ => throw InternalCompilerException("Unexpected JSON format.", SourceLocation.Unknown)
+    }
+
+    m.toMap
+  }
+
+  /**
+    * Parses a JSON file of the form:
+    *
+    * {{{
+    * {
+    *   "methods": {
+    *     "java.lang.System::exit": "Sys"
+    *   }
+    * }
+    * }}}
+    *
+    * Note: The effect set applies to *ALL* constructors of the class.
+    */
+  private def loadMethodEffs(): Map[Method, Set[Symbol.EffectSym]] = {
+    val data = LocalResource.get(MethodEffsPath)
+    val json = parse(data)
+
+    val m = json \\ "methods" match {
+      case JObject(l) => l.flatMap {
+        case (classNameAndMethod, JString(s)) =>
+          val cc = classNameAndMethod.indexOf("::")
+          val className = classNameAndMethod.substring(0, cc)
+          val methodName = classNameAndMethod.substring(cc + 2)
```

Comment:
crude, but ok

Reply:
Yeah. Any idea?

Reply:
I used this specific syntax because its valid Java syntax. I learned that yesterday.. You can write `forEach(System.out::println)`.